official/recommendation/data_preprocessing.py

```diff
@@ -644,10 +661,43 @@ def input_fn(params):
   return input_fn, record_dir, batch_count
 
 
+_SYNTHETIC_BATCHES_PER_EPOCH = 2000
```

Comment:
We know that the number of train batches should just be  about `(1 or 20 million) * (1 + num_neg) / train_batch_size` and eval will just be about `num_users * (1 + num_eval_neg) / eval_batch_size`. Can you just mimic the real behavior?

Reply:
Running for more batches is convenient to reduce noise. Added a comment stating this.